### PR TITLE
Refactor Code Part 2

### PR DESCRIPTION
**Pull request by Christian-Oleson**
Keeping things small as I am getting to this when I have a few minutes. It'll keep the code review smaller for you as well, and we can prevent breaking things without a ton of unit tests.

### CODE REVIEW

File: src/prospectScraper/MockDraftPick.cs
Comment (zo0o0ot) [inline src/prospectScraper/MockDraftPick.cs:47]: I'd prefer if this comment was kept, since this information is useful, and changes from year to year.

File: src/prospectScraper/MockDraftPick.cs
Comment (zo0o0ot) [inline src/prospectScraper/MockDraftPick.cs:102]: This comment is also useful and should be kept.

**Review by zo0o0ot — CHANGES_REQUESTED** [review]
Please preserve the two comments I mentioned.  Other than that, things look good.

File: src/prospectScraper/MockDraftPick.cs
Comment (Christian-Oleson) [reply src/prospectScraper/MockDraftPick.cs:102]: Ah, my bad. I tend to have an issue with comments, so I can be aggressive removing them from codebases. I wonder if there is a way to make this information useful, within the console app, without making it a comment? 🤔

File: src/prospectScraper/MockDraftPick.cs
Comment (Christian-Oleson) [reply src/prospectScraper/MockDraftPick.cs:47]: Reverted

**Comment by Christian-Oleson** [comment]
> Please preserve the two comments I mentioned. Other than that, things look good.

Done. 😃

**Comment by zo0o0ot** [comment]
Crap.  I'm sorry about this, but I just added a new parameter to handle #32 in #35 and it looks like I introduced some conflicts.

**Comment by zo0o0ot** [comment]
Would you be willing to port over my changes to the appropriate new files that you've created? I ended up having to change more than I expected, since the date parsing was moved into a mock draft specific method and left inline for the big board scraping.  I ended up needing to make two separate methods, one that returned a DateTime and the other returning a string since apparently the way I create the file name for the csv files wasn't consistent between the mock draft and the big board.

**Comment by Christian-Oleson** [comment]
> Would you be willing to port over my changes to the appropriate new files that you've created? I ended up having to change more than I expected, since the date parsing was moved into a mock draft specific method and left inline for the big board scraping. I ended up needing to make two separate methods, one that returned a DateTime and the other returning a string since apparently the way I create the file name for the csv files wasn't consistent between the mock draft and the big board.

I think I just moved over everything needed? If not, please let me know. That was a nasty set of merge issues. Once we get everything properly refactored, we can put the unit tests in that will catch behavior changes if I introduce them to the project.

**Comment by Christian-Oleson** [comment]
> Crap. I'm sorry about this, but I just added a new parameter to handle #32 in #35 and it looks like I introduced some conflicts.

And don't worry, these things happen. That's the reason I'm trying to keep the PRs smaller though, so we can get the changes in rather quickly, so we don't have to deal with merge issues. 😄

File: src/prospectScraper/Program.cs
Comment (zo0o0ot) [inline src/prospectScraper/Program.cs]: this line needs to be removed.  It references the old generic variable "s".

File: src/prospectScraper/ProspectScraper.cs
Comment (zo0o0ot) [inline src/prospectScraper/ProspectScraper.cs]: This comment can be deleted.

File: src/prospectScraper/Program.cs
Comment (zo0o0ot) [inline src/prospectScraper/Program.cs:260]: This method is new. It needs to be in ProspectScraper.cs. 

It replaces the functionality of GetDraftDate() and FormatDraftDate()

File: src/prospectScraper/Program.cs
Comment (zo0o0ot) [inline src/prospectScraper/Program.cs:276]: I see that this method has been replaced with an inline `.ToString("yyyy-MM-dd");`  I'm cool with that.  Mostly I was trying to salvage the method that had been previously created, and it isn't really necessary.

File: src/prospectScraper/Program.cs
Comment (Christian-Oleson) [reply src/prospectScraper/Program.cs]: Woah, something did not push right in this merge resolution. Taking a look now...

File: src/prospectScraper/ProspectScraper.cs
Comment (zo0o0ot) [inline src/prospectScraper/ProspectScraper.cs:39]: Here's the method, but it does not exist in ProspectScraper.cs

File: src/prospectScraper/Program.cs
Comment (Christian-Oleson) [reply src/prospectScraper/Program.cs]: but this is done

File: src/prospectScraper/Program.cs
Comment (Christian-Oleson) [reply src/prospectScraper/Program.cs:260]: Yeah, this got hosed in the merge conflict resolution. I'm adding it back.

File: src/prospectScraper/ProspectScraper.cs
Comment (zo0o0ot) [inline src/prospectScraper/ProspectScraper.cs:98]: This method call can be replaced with 
`string draftDate = mockDraftDate.ToString("yyyy-MM-dd");`

File: src/prospectScraper/Program.cs
Comment (Christian-Oleson) [reply src/prospectScraper/Program.cs:276]: I'll add more refactoring after we get this one merged in and fixed up that should make these classes smaller and more readable. 😄

File: src/prospectScraper/ProspectScraper.cs
Comment (Christian-Oleson) [reply src/prospectScraper/ProspectScraper.cs:39]: Fixed up. It pulled a Houdini.

File: src/prospectScraper/ProspectScraper.cs
Comment (zo0o0ot) [inline src/prospectScraper/ProspectScraper.cs]: This `FormatScrapedDate` method can be replaced with:
```suggestion
            string draftDate = mockDraftDate.ToString("yyyy-MM-dd");
```
`string draftDate = mockDraftDate.ToString("yyyy-MM-dd");`

**Comment by Christian-Oleson** [comment]
@zo0o0ot ,
You should see the new changes reflected shortly. I think I forgot to stage some of the changes, and then discarded some other changes, in the attempt to merge things up. Unit tests ran initially, but because I didn't push everything, it left this in a non-working state. Unit tests are passing again, everything is building, and fingers crossed we should be good to go here.

File: src/prospectScraper/ProspectScraper.cs
Comment (zo0o0ot) [inline src/prospectScraper/ProspectScraper.cs]: This method is no longer needed.

File: src/prospectScraper/ProspectScraper.cs
Comment (zo0o0ot) [inline src/prospectScraper/ProspectScraper.cs:218]: This method is no longer needed.

**Review by zo0o0ot — CHANGES_REQUESTED** [review]
See comments for the suggested changes, including the method changes.

File: src/prospectScraper/ProspectScraper.cs
Comment (Christian-Oleson) [reply src/prospectScraper/ProspectScraper.cs:218]: This is referenced in the tests. I need to unfurl that at some point. We want to remove as much as possible static classes and fields, as they tightly couple the code.

**Comment by Christian-Oleson** [comment]
> See comments for the suggested changes, including the method changes.

Looks like I was about 1 minute behind you. Let me know if there is anything else we need to hit. 😄

**Comment by Christian-Oleson** [comment]
> > See comments for the suggested changes, including the method changes.
> 
> Looks like I was about 1 minute behind you. Let me know if there is anything else we need to hit. 😄

@zo0o0ot ,
Also went in and refactored a tad more once I cleaned that up. I think this is starting to take form. 😄

File: src/prospectScraper/ProspectScraper.cs
Comment (Christian-Oleson) [inline src/prospectScraper/ProspectScraper.cs:33]: This is the one that I will eventually get to. We can eventually use polymorphism to allow any sort of draft, and we can find a way to make this scale based on that draft type. Prefer the MLB? NHL? NBA? Well, maybe, just maybe someday, that will be possible.

Comment (zo0o0ot) [reply src/prospectScraper/ProspectScraper.cs:33]: Check out #25 as there is some good stuff there, but I haven't merged it yet because we're still moving stuff around all over the place.

Comment (Christian-Oleson) [reply src/prospectScraper/ProspectScraper.cs:33]: Awesome! I'll stop moving things for now so we can get some of this in then. 👍

**Review by zo0o0ot — APPROVED** [review]
Looks good.

**Pull request merged**
